src/rgw/rgw_rados.cc

```diff
@@ -6780,12 +6780,13 @@ int RGWRados::swift_versioning_restore(RGWObjectCtx& obj_ctx,
  * mtime: if non-NULL, writes the given mtime to the bucket storage
  * attrs: all the given attrs are written to bucket storage for the given object
  * exclusive: create object exclusively
+ * completeMultipart: complete multipart or not
  * Returns: 0 on success, -ERR# otherwise.
  */
 int RGWRados::Object::Write::_do_write_meta(uint64_t size, uint64_t accounted_size,
                                            map<string, bufferlist>& attrs,
                                            bool assume_noent, bool modify_tail,
-                                           void *_index_op)
+                                           void *_index_op, bool completeMultipart)
```

Comment:
@gaosibei is it possible accounted_size could be corrected by the call path leading to _do_write_meta?

Reply:
I think that the accounted_size could not be corected outside of the _do_write_meta. Because, the accounted_size is  both used in quota status and index_op.

at the rgw_rados.cc:6940 (_do_write_meta)
 r = index_op->complete(poolid, epoch, size, accounted_size,...)